src/core/server/saved_objects/migrationsv2/integration_tests/migration_7_13_0_transform_failures.test.ts

```diff
@@ -0,0 +1,165 @@
+}
+
+describe('migration v2', () => {
+  let esServer: kbnTestServer.TestElasticsearchUtils;
+  let root: Root;
+
+  beforeAll(async () => {
+    await removeLogFile();
+  });
+
+  afterAll(async () => {
+    if (root) {
+      await root.shutdown();
+    }
+    if (esServer) {
+      await esServer.stop();
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+  });
+
+  it('migrates the documents to the highest version', async () => {
+    const { startES } = kbnTestServer.createTestServers({
+      adjustTimeout: (t: number) => jest.setTimeout(t),
+      settings: {
+        es: {
+          license: 'basic',
+          // example of original 'bar' SO:
+          // {
+          //  type: 'bar',
+          //  bar: {
+          // name: 'Mary', // may be missing
+          // surname: 'Doe, // may be missing
+          // age: 25, // may be missing
+          // old: false // may be missing
+          // },
+          //  migrationVersion: {
+          //    bar: '7.13.0',
+          //  },
+          // },
+          // SO that will fail transformation:
+          // {
+          //  type: 'space',
+          //  space: {},
+          // },
+          //
+          //
+          dataArchive: Path.join(
+            __dirname,
+            'archives',
+            '7.13.0_with_corrupt_and_transform_failures_docs.zip'
+          ),
+        },
+      },
+    });
+
+    root = createRoot();
+
+    esServer = await startES();
+    const coreSetup = await root.setup();
+
+    coreSetup.savedObjects.registerType({
+      name: 'foo',
+      hidden: false,
+      mappings: {
+        properties: {},
+      },
+      namespaceType: 'agnostic',
+      migrations: {
+        '7.14.0': (doc) => doc,
+      },
+    });
+    try {
+      await root.start();
+    } catch (err) {
+      const errorMessage = err.message;
+      expect(
+        errorMessage.startsWith(
+          'Unable to complete saved object migrations for the [.kibana] index: Migrations failed. Reason: Corrupt saved object documents: '
+        )
+      ).toBeTruthy();
+      expect(
+        errorMessage.endsWith(' To allow migrations to proceed, please delete these documents.')
+      ).toBeTruthy();
+
+      const expectedCorruptDocIds = [
+        'P2SQfHkBs3dBRGh--No5',
+        'QGSZfHkBs3dBRGh-ANoD',
+        'QWSZfHkBs3dBRGh-hNob',
+        'QmSZfHkBs3dBRGh-w9qH',
+        'Q2SZfHkBs3dBRGh-9dp2',
+        'one',
+        'two',
+      ];
+      const corruptDocIds = errorMessage
+        .split(' Corrupt saved object documents: ')[1]
+        .split(' Transformation errors')[0]
+        .split(',');
+      expect(corruptDocIds.length).toBeGreaterThan(5);
+      expect(corruptDocIds).toEqual(expectedCorruptDocIds);
+
+      const transformErrorInstances = errorMessage
+        .split(' Transformation errors: ')[1]
+        .split('Error')
+        .filter((instance: string) => !instance.startsWith('Error'));
```

Comment:
I'm open to suggestions and examples on how to make this more elegant. Each error we have within the string has the following structure:
```
space:default: Document \"default\" has property \"space\" which belongs to a more recent version of Kibana [6.6.0]. The last known version is [undefined]\n Error: Document \"default\" has property \"space\" which belongs to a more recent version of Kibana [6.6.0]. The last known version is [undefined]\n    at find (/Users/christianeheiligers/Projects/kibana/src/core/server/saved_objects/migrations/core/document_migrator.ts:727:18)\n    at Array.find (<anonymous>)\n    at nextUnmigratedProp (/Users/christianeheiligers/Projects/kibana/src/core/server/saved_objects/migrations/core/document_migrator.ts:717:21)\n    at applyMigrations (/Users/christianeheiligers/Projects/kibana/src/core/server/saved_objects/migrations/core/document_migrator.ts:487:18)\n    at DocumentMigrator.transformDoc (/Users/christianeheiligers/Projects/kibana/src/core/server/saved_objects/migrations/core/document_migrator.ts:434:22)\n    at transform (/Users/christianeheiligers/Projects/kibana/src/core/server/saved_objects/migrations/core/document_migrator.ts:246:53)\n    at Immediate.<anonymous> (/Users/christianeheiligers/Projects/kibana/src/core/server/saved_objects/migrations/core/migrate_raw_docs.ts:152:19)\n    at processImmediate (internal/timers.js:461:21)/n
```

Reply:
we also can assert against sub-string in the error message